src/reconcile/rec_visibility.c

```diff
@@ -429,13 +436,24 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
      */
     if (!F_ISSET(r, WT_REC_LOOKASIDE | WT_REC_UPDATE_RESTORE))
         WT_ERR(__wt_set_return(session, EBUSY));
-    if (list_uncommitted && !F_ISSET(r, WT_REC_UPDATE_RESTORE))
+
+    /*
+     * If we have a list with an uncommitted update on it we still want to write out to the
+     * lookaside file in checkpoint.
+     */
+    if (list_uncommitted && !F_ISSET(r, WT_REC_UPDATE_RESTORE) && !F_ISSET(r, WT_REC_CHECKPOINT))
```

Comment:
Why we don't do that for prepared updates?

Reply:
OK, I think I missed this in my previous change. We should check prepared updates here as well.

Reply:
Can we merge list_prepared and list_uncommitted, in the scenario we add list_prepared to this list its used identically to list_uncommitted.

Reply:
According to Vamsi, we probably need to insert prepared updates into las in the future. I think it's better to keep them separate?

Reply:
Spoke offline, agreed to merge the two local values but keep the values in the reconciliation structure.